### PR TITLE
Rework dockerfile

### LINKED ISSUE #445: Dockerfile is not granular enough
It would be a lot easier to work with the Dockerfile if the part where it downloads and compiles the Go toolchain was a seperate RUN step from the compilation of prometheus, since otherwise it attempts to download and build Go every. single. time. you. run. docker. build.

### PR DESCRIPTION
**Pull request by discordianfish**
Instead of using make to install all dependencies, this uses the golang
base image. This way those common base layers can be shared across other
golang based images.
This also stops using the Makefile to install the dependencies because
we would have to install all dependencies everytime something changes.

This fixes #445

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,27 @@
-FROM       ubuntu:13.10
+FROM       golang:latest
```
Comment (juliusv) [inline Dockerfile]: I think we should keep the Go versions in sync between the Makefile and the Dockerfile.

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
+RUN  godep restore && go get -d
+RUN  ./utility/embed-static.sh web/static web/templates | gofmt > web/blob/files.go
+
+RUN  go build -ldflags " \
+       -X main.buildVersion  $(cat VERSION) \
+       -X main.buildRevision $(git rev-parse --short HEAD) \
+       -X main.buildBranch   $(git rev-parse --abbrev-ref HEAD) \
+       -X main.buildUser     root \
+       -X main.buildDate     $(date +%Y%m%d-%H:%M:%S) \
+       -X main.goVersion     $GOLANG_VERSION \
```
Comment (juliusv) [inline Dockerfile]: Where does `$GOLANG_VERSION` get set?

**Comment by juliusv** [comment]
Ok, this duplicates some logic from the Makefile, but if it fixes significant other pains, :+1: 

@swsnider - what do you think?

File: Dockerfile (hunk 2 above)
Comment (discordianfish) [inline Dockerfile]: In the base image (that FROM <image> stuff)

Comment (juliusv) [inline Dockerfile]: Aha!

**Comment by discordianfish** [comment]
@juliusv I've changed the base image to golang:1.4. We just need to make sure to update this along the way then.

**Pull request merged**